test/Microsoft.NET.Build.Tests/AppHostTests.cs

```diff
@@ -509,5 +507,62 @@ private static bool IsPE32(string path)
                 return reader.PEHeaders.PEHeader.Magic == PEMagic.PE32;
             }
         }
+
+        // Reads the Mach-O load commands and returns true if an LC_CODE_SIGNATURE command is found, otherwise returns false
+        static bool HasMachOSignatureLoadCommand(FileInfo file)
```

Comment:
Doesn't matter for this change, but we may want this in a shared test helper so that we can add cross-platform checks for single-file publish with signing after https://github.com/dotnet/runtime/pull/110417 goes in.